megatron/model/norms.py

```diff
@@ -57,6 +57,7 @@ def __init__(self, dim, p=-1.0, eps=1e-8, bias=False):
             self.register_parameter("offset", self.offset)
 
     def forward(self, x):
+        dtype = x.dtype
```

Comment:
In the future, we should add support for the triton RMSNorm kernel introduced by mamba. Noting here and adding a TODO for later.

Reply:
https://github.com/state-spaces/mamba/blob/v1.2.0/mamba_ssm/ops/triton/layernorm.py